src/protocol/ascii.rs

```diff
@@ -41,11 +41,98 @@ impl fmt::Display for StoreCommand {
         }
     }
 }
+
+struct CappedLineReader<C> {
+    inner: C,
+    filled: usize,
+    buf: Box<[u8]>,
+}
+
+fn get_line(buf: &[u8]) -> Option<usize> {
+    for (i, r) in buf.iter().enumerate() {
+        if *r == b'\r' {
+            if buf.get(i + 1) == Some(&b'\n') {
+                return Some(i + 2);
+            }
+        }
+    }
+    None
+}
+
+impl<C: Read> CappedLineReader<C> {
+    fn with_capacity(capacity: usize, inner: C) -> Self {
+        let buf = vec![0x0; capacity];
```

Comment:
If we are using just one capacity, can we hard code it here to avoid dynamic allocations?

Reply:
Yes, I guess we can do this. I'll change.